src/core/Rule.ml

```diff
@@ -358,6 +358,23 @@ and extract_reduction = Separate | Concat [@@deriving show]
 and extract_transform = NoTransform | Unquote | ConcatJsonArray
 [@@deriving show]
 
+(*****************************************************************************)
+(* Secrets mode *)
+(*****************************************************************************)
+
+(*TODO: actualy figure this out*)
+type secrets_spec = {
+  (* secrets: *)
+  secrets : formula list;
+  (* request: *)
+  url : string;
+  method_ : string;
```

Comment:
Can we use a more precise type?
The http libraries usually define this like Get | Post | ...

Reply:
Changed it to `meth : [`GET | `POST]`.